### DISCUSSION TITLE
Need help with GEOSX installation on Git Bash

### DISCUSSION THREAD

**Discussion by prashantkgp01** [GEOS-DEV/GEOS#2558] [Q&A]
Hello everyone,

I'm currently facing some challenges with installing GEOSX on Git Bash in windows OS,  I require guidance regarding the configuration process and the selection of the appropriate '.cmake' file in the 'GEOS/host-configs' directory.

Here are my specific questions:

Selection of '.cmake' file: When navigating to the 'GEOS/host-configs' directory using the command "cd GEOS/host-configs", which type of '.cmake' file should I choose for my configuration? 

Configuration changes: What modifications should I make within this file once I have selected the appropriate '.cmake' file? 

Compilation and installation: Could you provide more detailed instructions or insights on proceeding with the compilation and installation steps once the configuration is complete? Any tips or additional resources would be highly appreciated.

Thank you for your time

**Comment by klevzoff** [comment]
Hi @prashantkgp01,

By Git Bash do you mean the console that comes with installing Git natively on Windows? If so, GEOS does not support building directly on Windows. We recommend installing [WSL](https://learn.microsoft.com/en-us/windows/wsl/install) and building GEOS and all its dependencies from the WSL terminal.

If you choose to do that, you may want to:
1. Install system prerequisites as listed [here](https://geosx-geosx.readthedocs-hosted.com/en/latest/docs/sphinx/buildGuide/Prerequisites.html) (note that this list is about to go out-of-date, for example we will start requiring minimum CMake 3.23.1 and c++17 supporting compilers).
2. Use [environment.cmake](https://github.com/GEOS-DEV/GEOS/blob/develop/host-configs/environment.cmake) host-config as a start. You will need to first set environmental variables `CC`, `CXX`, `MPICC`, `MPICXX`, `MPIEXEC` to their respective locations on the system (most likely, `/usr/bin/[gcc/g++/mpicc/mpicxx]`, as well as `ENABLE_HYPRE=ON`.
3. Build third-party libraries following these [instructions](https://geosx-geosx.readthedocs-hosted.com/en/latest/docs/sphinx/buildGuide/Dependencies.html#building-bundled-dependencies).
4. Build GEOS following these [steps](https://geosx-geosx.readthedocs-hosted.com/en/latest/docs/sphinx/buildGuide/BuildProcess.html).

Let us know how it goes!

**Reply by prashantkgp01** [reply]
Hi @klevzoff,

Thank you for your kind response...

Regarding your suggestion to install WSL (Windows Subsystem for Linux), I would like to clarify if it is necessary for my case. I already have CentOS(version 2.6.32-754.el6) installed on a virtual machine using VirtualBox/VMware Workstation. Can I proceed with the installation of GEOSX on CentOS within the virtual machine environment without installing WSL?

Additionally, I encountered an issue while trying to install GEOSX on CentOS. Despite having installed Git on CentOS, the Git command is not found. how to resolve this issue and ensure that the Git command is recognized within the CentOS virtual machine.

Thank you once again for your assistance. I look forward to your response.

**Reply by klevzoff** [reply]
Yes, you can build GEOS in a virtual machine, although I can't comment on performance when running in a VM vs WSL as I have done neither personally. Perhaps you want to make sure you allow the VM to use all of your CPU cores at least.

How did you install Git? I believe if you run `yum install git` you should have the `git` command available. That said, you're running CentOS 6 which I haven't worked with, so I can't help much with this. Would you consider making a separate VM image for GEOS based on a more recent CentOS version (8 or 9)?
